webapp/channels/src/packages/mattermost-redux/src/selectors/entities/groups.ts

```diff
@@ -117,28 +117,30 @@ export function searchAssociatedGroupsForReferenceLocal(state: GlobalState, term
     return filteredGroups;
 }
 
-export function getAssociatedGroupsForReference(state: GlobalState, teamId: string, channelId: string): Group[] {
+export function getAssociatedGroupsForReference(state: GlobalState, teamId: string, channelId?: string): Group[] {
```

Comment:
Could you actually have this use `createSelector`? It's returning a new array each time if the team or channel is  group-constrained